src/modules/error.rs

```diff
@@ -0,0 +1,20 @@
+use std::fmt::{Debug, Formatter, Result};
+
+#[derive(PartialEq)]
+pub struct Error {
+    msg: String,
+}
+
+impl Error {
+    pub fn from(msg: &str) -> Error {
+        Error { msg: String::from(msg) }
+    }
+}
+
+impl Debug for Error {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        f.debug_struct("RegexError")
+            .field("Internal Error", &self.msg)
```

Comment:
We emit an error if the input contains an illegal character. I feel that's not really an internal error.

Reply:
Fair enough, changed this to a more general name.